src/sql/platform/connection/common/connectionProfile.ts

```diff
@@ -41,6 +43,12 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
 			this.saveProfile = model.saveProfile;
 			this._id = model.id;
 			this.azureTenantId = model.azureTenantId;
+			const options = this.capabilitiesService.getCapabilities(model.providerName).connection.connectionOptions;
+			let appNameOption = options.find(option => option.specialValueType === ConnectionOptionSpecialType.appName);
```

Comment:
The appname is added by the dialog, which means in cases where we don't hit the dialog the profile won't match for matching cases.